test/optimization/test_admm_miskp.py

```diff
@@ -1,199 +1,95 @@
-    print("c1")
-    print(c1)
-    print("c1 shape")
-    print(c1.shape)
-
-    print("A0")
-    print(A0)
-    print("A0 shape")
-    print(A0.shape)
-    print("b0")
-    print(b0)
-    print("b0 shape")
-    print(b0.shape)
-
-    print("A1")
-    print(A1)
-    print("A1 shape")
-    print(A1.shape)
-    print("b1")
-    print(b1)
-    print("b1 shape")
-    print(b1.shape)
-
-    print("A4")
-    print(A4)
-    print("A4 shape")
-    print(A4.shape)
-    print("b3")
-    print(b3)
-    print("b3 shape")
-    print(b3.shape)
-
-    print("A2")
-    print(A2)
-    print("A2 shape")
-    print(A2.shape)
-    print("A3")
-    print(A3)
-    print("A3 shape")
-    print(A3.shape)
-    print("b2")
-    print(b2)
-    print("b2 shape")
-    print(b2.shape)
-
-    print("End of Claudio's implementation")
-
-
-def get_instance_params():
-    """
-    Get parameters for a  Mixed Integer Setup Knapsack Problem (MISKP) instance.
-
-    :return:
-    """
-    #
-
-    K = 2
-    T = 10
-    P = 45.10
-    S = np.asarray([75.61, 75.54])
-
-    D = np.asarray([9.78, 8.81, 9.08, 2.03, 8.9, 9, 4.12, 8.16, 6.55, 4.84, 3.78, 5.13,
-                    4.72, 7.6, 2.83, 1.44, 2.45, 2.24, 6.3, 5.02]).reshape((K, T))
-
-    C = np.asarray([-11.78, -10.81, -11.08, -4.03, -10.90, -11.00, -6.12, -10.16, -8.55, -6.84,
-                    -5.78, -7.13, -6.72, -9.60, -4.83, -3.44, -4.45, -4.24, -8.30, -7.02]).reshape((K, T))
-
-    n = K  # number of binary dec vars x0 (and z)
-    m = K * T  # number of continuous dec vars u
-    # size of x1 is then n+m
-
-    # Parameters of the Quadratic Programming problem in the standard form for ADMM.
-    ## Objective function
-    Q0 = np.zeros((n, n))
-    c0 = S.reshape(n)
-
-    A0 = np.zeros((1, n))  # we set here one dummy equality constraint
-    b0 = np.zeros(1)  # we set here one dummy equality constraint
-
-    Q1 = np.zeros((m, m))
-    c1 = C.reshape(m)
-
-    ## Constraints
-    # we set here one dummy A1 x0 \leq b1 constraint
-    A1 = np.zeros((1, n))
-    b1 = np.zeros(1)
-
-    # A_2 z + A_3 u \leq b_2 -- > - y_k <= x_{k,t}
-    A2 = np.zeros((m, n))
-    for i in range(K):
-        A2[T * i:T * (i + 1), i] = - np.ones(T)
-    A3 = np.eye(m)
-    b2 = np.zeros(m)
-
-    # A_4 u <= b_3 --> sum_k sum_t D_{kt} x_{kt} \leq P, -x_{kt} \leq 0
-    block1 = D.reshape((1, m))
-    block2 = -np.eye(m)
-    A4 = np.block([[block1], [block2]])
-    b3 = np.hstack([P, np.zeros(m)])
-
-    __dump_arrays(Q0, Q1, c0, c1, A0, b0, A1, b1, A4, b3, A2, A3, b2)
-
-    return K, T, P, S, D, C
+class TestADMMOptimizerMiskp(QiskitOptimizationTestCase):
+    """ADMM Optimizer Tests based on Mixed-Integer Setup Knapsack Problem"""
 
+    def setUp(self):
+        super().setUp()
 
+    def test_admm_optimizer_miskp_eigen(self):
+        """ ADMM Optimizer Test based on Mixed-Integer Setup Knapsack Problem using NumPy eigen optimizer"""
+        K, T, P, S, D, C = self.get_problem_params()
+        miskp = Miskp(K, T, P, S, D, C)
```

Comment:
These names will raise an error in pylint, but you can just do this
```suggestion
        miskp = Miskp(*self.get_problem_params())
```
which uses cool python syntax 😉

Reply:
Accepted